code/cfile/cfile.cpp

```diff
@@ -1863,3 +1868,44 @@ int cfile_get_path_type(const SCP_string& dir)
 
 	return CF_TYPE_INVALID;
 }
+
+// find the last separator (if any) in the path sequence
+const char *find_last_separator(const char *path_str)
+{
+	const char *separators, *last_separator = nullptr, *ch;
+
+#ifdef SCP_UNIX
+	separators = "/";
+#else
+	separators = "/\\:";
+#endif
+
+	// find the last separator (if any) in the path sequence
+	ch = path_str - 1;
+	while ((ch = strpbrk(ch + 1, separators)) != nullptr) {
+		last_separator = ch;
+	}
+
+	return last_separator;
```

Comment:
You may could just do this instead (note the 2 r's, for reverse):

`return strrchr(ch, DIR_SEPARATOR_CHAR);`

which will return a ptr to the last directory separator, or NULL. No need to specify the separators then either

Reply:
I could, except that the cfile code explicitly specifies more than one possible separator char.

Reply:
On windows it's looking for something like `c:\` or `c:/` to indicate a full path. In all other places in the code we specifically look for DIR_SEPARATOR_*, so I don't think changing this would break anything or else we should have already noticed.

May want to give it a test though. I don't have a windows box handy to confirm that it would be okay.

Reply:
Ok, seems to work.